### ISSUE TITLE
5.x: Add upload file dialog

### ISSUE THREAD

**Issue by petrkalina** [gollum/gollum#1529]
Currently, the only way to add images from local computer is via drag-and-drop - the same as github. 

It would be nice to have this functionality available under the "add image" button in the editor just to make it simple for the user to figure out how gollum works with images without having to experiment. 

Maybe just adding a `Drag files here` drop area in this dialog would be sufficient start - as same finctionality is already supported by the editor..

**Comment by dometto** [comment]
PRs welcome, I won't be working on this myself

**Comment by bartkamphorst** [comment]
> the only way to add images from local computer is via drag-and-drop - the same as github.

Actually, Github wikis don't support drag-and-drop uploads, nor do they have an "add image" button in their editor.

**Comment by greg-tbca** [comment]
How does one add a file to the repo using the Gollum interface?

**Comment by dometto** [comment]
I think @petrkalina meant "just as in github comment fields" :)

**Comment by petrkalina** [comment]
You can always copy the file to the file tree by i.e. scp and then use git
to commit it to the underlying repo .. so that gollum sees it. This request
was about making this possible also via the UI, where there is no way to
upload images other than drag-and-drop that I know of - which makes working
with the wiki less intuitive and also dependent on browser/OS support for
drag and drop..

**Comment by hmasila** [comment]
I'd like to follow up on this. Has anyone been successful at implementing the copy/paste feature for images

**Comment by bartkamphorst** [comment]
@hmasila As far as I know, this hasn't been moved along by anyone. Just to be sure, the OP requested a file upload dialog, but copy/paste would certainly also be a welcome addition. Would you be willing to work on this?

**Comment by hmasila** [comment]
I looked into it and it seemed that the functionality (copy/paste) is missing in the ace editor so the solution will have to be implemented in ace then added to gollum as part of ace lib files

**Comment by MrOggy85** [comment]
Workaround:
You can also just base64 encode your image and then add it with <img> tag into a separate .md file (e.g. `my-image.md`)
Then you can use include e.g. `[[include:my-image.md]]` in the page you want it.

This would basically let you create local "image files" (in a MD file) that you can reference.

**Comment by psy0rz** [comment]
yes this is the only thing missing: We rely heavily on editting by phone and uploading pictures by phone. Currently  this is basically impossible to do.

Leafwiki is much better at this, but i like the git based nature of gollum.

**Comment by dometto** [comment]
Is ace still unable to do this?
